examples/occlusion.html

```diff
@@ -0,0 +1,392 @@
+  model = new THREE.Mesh(new THREE.TeapotBufferGeometry(MODEL_SIZE_IN_METERS, 15),
+    new THREE.MeshLambertMaterial( {color: 0x888888 } ));
+  model.geometry.translate(0, MODEL_SIZE_IN_METERS / 2, 0);
+  model.position.set(0, 0, -2);
+  scene.add(model);
+  // Everything related to the point cloud that uses its own scene
+  // to render it specifically for occlusion
+  pointsScene = new THREE.Scene();
+  // Use a custom shader for the points so the points that are considered
+  // to be in the same plane as the picking plane are discarded.
+  // This way, the model will not be occluded by the points in the
+  // same plane as the picking plane.
+  var pointsMaterial = new THREE.RawShaderMaterial({
+    uniforms: {
+      size: { value: 30 },
+      opacity: { value: 0.1 },
+      color: { value: new THREE.Color(0xffffff) },
+      plane: { value: new THREE.Vector4() },
+      distance: { value: 0.05 }
+    },
+    vertexShader: points_vertexShader,
+    fragmentShader: points_fragmentShader
+  });
+  pointCloud = new VRPointCloudWrapper(vrDisplay, true);
+  points = new THREE.Points(pointCloud.getBufferGeometry(),
+    pointsMaterial);
+  // Points are changing all the time so calculating the frustum culling
+  // volume is not very convenient.
+  points.frustumCulled = false;
+  pointsScene.add(points);
+  // Control the perspective camera using the VR pose.
+  vrControls = new THREE.VRControls(camera);
+  // Add some lighting
+  var directionalLight = new THREE.DirectionalLight( 0xffffff, 0.5 );
+  directionalLight.position.set( 0, 1, 0);
+  scene.add(directionalLight);
+
+  // Control the resizing of the window to correctly display the scene.
+  window.addEventListener( 'resize', onWindowResize, false );
+  // Wherever the user clicks in the screen, place the model.
+  renderer.domElement.addEventListener("click", function(event) {
+    pos.x = event.pageX / window.innerWidth;
+    pos.y = event.pageY / window.innerHeight;
+    picking();
+  });
+  // Control the rotation of the model
+  var lastX = 0;
+  renderer.domElement.addEventListener("touchstart", function(event) {
+    if (event.changedTouches.length > 0)
+      lastX = event.changedTouches[0].pageX;
+  });
+  renderer.domElement.addEventListener("touchmove", function(event) {
+      if (event.changedTouches.length > 0 && vrDisplay) {
+        var x = event.changedTouches[0].pageX;
+        var diffX = x - lastX;
+        lastX = x;
+        model.rotateX(THREE.Math.degToRad(diffX));
+      }
+  });
+
+  update();
+}
+function picking() {
+  if (vrDisplay) {
+    var hits =
+        vrDisplay.hitTest(pos.x, pos.y);
+    if (hits && hits.length > 0) {
+      var hit = hits[0];
+      // Orient the cone model at the position and orientation of the hit.
+      THREE.ARUtils.placeObjectAtHit(model, hit, 1, true);
+      points.material.uniforms.plane.value.fromArray(
+          [hit.modelMatrix[4], hit.modelMatrix[5], hit.modelMatrix[6], hit.modelMatrix[7]]);
+      /*
```

Comment:
This chunk of comments can probably be removed

Reply:
done